app/scripts/controllers/detect-tokens.test.js

```diff
@@ -559,4 +566,42 @@ describe('DetectTokensController', function () {
     clock.tick(180000);
     sandbox.assert.notCalled(stub);
   });
+
+  it('should poll on the correct interval by networkClientId', async function () {
+    jest.useFakeTimers();
+    const controller = new DetectTokensController({
+      messenger: getRestrictedMessenger(),
+      preferences,
+      network,
+      tokensController,
+      assetsContractController,
+      disableLegacyInterval: true,
+      interval: 1000,
+      getNetworkClientById: () => ({
+        configuration: {
+          chainId: '0x1',
+        },
+        provider: {},
+        blockTracker: {},
+        destroy: () => {
+          // noop
+        },
+      }),
+    });
+    const detectNewTokensSpy = jest
+      .spyOn(controller, 'detectNewTokens')
+      .mockResolvedValue('foo');
+    controller.startPollingByNetworkClientId('mainnet');
+    await Promise.all([jest.advanceTimersByTime(0), flushPromises()]);
+    expect(detectNewTokensSpy).toHaveBeenCalledTimes(1);
+    await Promise.all([jest.advanceTimersByTime(1000), flushPromises()]);
+    expect(detectNewTokensSpy).toHaveBeenCalledTimes(2);
+    expect(detectNewTokensSpy.mock.calls).toStrictEqual([
+      [{ chainId: '0x1' }],
+      [{ chainId: '0x1' }],
+    ]);
+
+    detectNewTokensSpy.mockRestore();
```

Comment:
no need to restore this since the controller is only used in the context of this scenario